packages/react-admin/src/AdminRoutes.js

```diff
@@ -81,69 +113,114 @@ export class AdminRoutes extends Component {
-                                render={route.props.render}
-                                children={route.props.children} // eslint-disable-line react/no-children-prop
+                                render={props => {
+                                    if (route.props.render) {
+                                        return route.props.render({
+                                            ...props,
+                                            title,
+                                        });
+                                    }
+                                    if (route.props.component) {
+                                        return createElement(
+                                            route.props.component,
+                                            {
+                                                ...props,
+                                                title,
+                                            }
+                                        );
+                                    }
+                                }}
                             />
                         ))}
-                    {Children.map(childrenToRender, child => (
-                        <Route
-                            path={`/${child.props.name}`}
-                            render={props =>
-                                cloneElement(child, {
-                                    context: 'route',
-                                    ...props,
-                                })}
-                        />
-                    ))}
-                    {dashboard ? (
-                        <Route
-                            exact
-                            path="/"
-                            render={routeProps => (
-                                <WithPermissions
-                                    authParams={{ route: 'dashboard' }}
-                                    {...routeProps}
-                                    render={props =>
-                                        React.createElement(dashboard, props)}
-                                />
-                            )}
-                        />
-                    ) : (
-                        childrenToRender[0] && (
-                            <Route
-                                exact
-                                path="/"
-                                render={() => (
-                                    <Redirect
-                                        to={`/${childrenToRender[0].props
-                                            .name}`}
+                <Route
+                    path="/"
+                    render={() =>
+                        createElement(appLayout, {
+                            children: (
+                                <Switch>
+                                    {customRoutes &&
+                                        customRoutes.map((route, index) => (
```

Comment:
you should filter out the custom routes without layout, which were already rendered above.